### ISSUE TITLE
CRF decoding method ?

### ISSUE THREAD

**Issue by ziodos** [wenwenyu/PICK-pytorch#93]
does the crf performs sequence labeling in token level or character level ?

**Comment by AtulKumar4** [comment]
character level

**Comment by AtulKumar4** [comment]
> does the crf performs sequence labeling in token level or character level ?

character level

**Comment by ziodos** [comment]
@AtulKumar4 thanks for the answer , can you please provide more informations about the format of IOB labels if the model perfroms sequence tagging in character level ?

**Comment by AtulKumar4** [comment]
@ziodos 
If you have a class name 'food' then iob format will like this:-
Example:- I will order french fries


o oooo ooooo B-food i-food i-food i-food i-food i-food B-food i-food i-food i-food i-food
Hope this will help.

**Comment by ziodos** [comment]
yes thank you so much, I still have confusion .. in the code they used both word and character embedding , so from what I understood the BiLSTM layer accecpts as input the character embedding plus the word embedding to where it belongs .. for example if we are treating the word "Total"  , the character "T" is represented by it's caracter embedding plus the word "Total" embedding, is that right ?

**Comment by AtulKumar4** [comment]
@ziodos 
if you read the decoder part in the paper. They have mentioned "Union layer receives the input X ∈ R
N×T ×dmodel
having variable length T generated from Encoder means each word have a different number of characters, then packs
padded input sequences and fill padding value at the end
of sequence yielding packed sequence Xˆ ∈ R
(N·T)×dmodel". 

Then this output will get combined with the graph embedding before feeding into CRF layer.
Example:-Rough Idea
Total -> (5+ pad)* (embedding vector length) Union (Graph embedding)
5+pad = maximum length of the sequence
